src/purview.ts

```diff
@@ -464,15 +465,16 @@ async function handleMessage(
       await Promise.all(deletePromises)
 
       wsState.connectionState = "connected"
-      // We check if `closing` is set because it could be the case that the
-      // "close" event fires just after the "connect" event (e.g., on page
-      // refresh), and the "close" event will see that the `wsState.roots` is an
-      // empty array due to the "connect" still being in progress. This would
-      // result in an incomplete cleanup of the previous connection's state.
-      // Hence, we check the `closing` flag and cleanup any existing state that
-      // the "closing" event could not cleanup if needed.
+      // Because both the "close" and "connect" events are async, we check if
+      // `closing` is set because it could be the case that the "close" event
+      // fires just after the "connect" event (e.g., on page refresh), and the
+      // "close" event will see that the `wsState.roots` is an empty array due
+      // to the "connect" event still being in progress. This would result in an
+      // incomplete clean up of the previous connection's state. Hence, we check
+      // the `closing` flag and clean up any existing state that the "closing"
+      // event could not cleanup if needed.
```

Comment:
Forgive me for the nits, but I think there was some miscommunication. On line 473, it should be cleanup because you're using it as a noun. On 474 and 475, it should be clean up because you're using it as a verb.

Reply:
Oops, I'll revise. My bad for over applying the feedback